### ISSUE TITLE
E2532 when Step Functions Pass state has Parameters key

### ISSUE THREAD

**Issue by bryanburke** [aws-cloudformation/cfn-python-lint#762]
*cfn-lint version: 0.17.0*

Using latest version of cfn-lint and updated spec files.

*Description of issue.*

Linting a CF template fails when a `AWS::StepFunctions::StateMachine`'s `DefinitionString` has a `Pass` state with the `Parameters` key.

Example template:

```yaml
AWSTemplateFormatVersion: 2010-09-09
Resources:
    ExampleStateMachine:
        Type: AWS::StepFunctions::StateMachine
        Properties:
            DefinitionString: |-
                {
                    "StartAt": "Pop Element from List",
                    "States": {
                        "Pop Element from List": {
                            "Type": "Pass",
                            "Parameters": {
                                "List.$": "$.List[1:]"
                            },
                            "End": true
                        }
                    }
                }
            RoleArn: redacted
```

Error:

```
E2532 State Machine Definition key (Parameters) for State (Pop Element from List) of Type (Pass) is not valid
example.yaml:6:13
```

However, the above `Pass` state conforms to the ASL [spec](https://states-language.net/spec.html#state-type-table) and [docs](https://docs.aws.amazon.com/step-functions/latest/dg/amazon-states-language-pass-state.html) and comes directly from the [Transfer Data Records sample project](https://docs.aws.amazon.com/step-functions/latest/dg/sample-project-transfer-data-sqs.html).

**Comment by fatbasstard** [comment]
Can you add the error you get when you try to deploy the template? This makes it more understandable and is a double check to make sure our rule is off, and not the documentation. 
(I assume the docs and the sample project is right, but I've seen code samples from AWS that "invalid" 😉 )

**Comment by bryanburke** [comment]
The template deploys successfully (after substituting a real role ARN, of course).
